src/app/App.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { MESSAGE_ACTIONS } from '../plugin/constants';
+import './styles/ui.css';
+
+export const App = () => {
+  const [availableThemeModes, setAvailableThemeModes] = useState(undefined);
+  const [selectedThemeModeId, selectThemeModeId] = useState(undefined);
+
+  useEffect(() => {
+    getThemeModes();
+
+    const handleMessage = (event) => {
+      const action = event.data?.pluginMessage?.action;
+      const payload = event.data?.pluginMessage?.payload;
+
+      if (action === MESSAGE_ACTIONS.DOWNLOAD) {
+        const anchor = document.createElement('a');
+        const blob = new Blob([JSON.stringify(payload)]);
+        anchor.href = URL.createObjectURL(blob);
+        anchor.download = 'theme.json';
+        anchor.click();
+      }
+
+      if (action === MESSAGE_ACTIONS.GET_THEME_MODES) {
+        setAvailableThemeModes(payload);
+      }
```

Comment:
I have not looked as this deeply, but, will it makes sense to use `useReducer` for this?

Reply:
if not, can we change it to use a `switch` instead?